tests/zexperimental/test_mcmc.py

```diff
@@ -103,6 +103,36 @@ def test_MALA_regression_warmup(target: cuqi.density.Density):
     sampler_new = cuqi.experimental.mcmc.MALANew(target, scale=1)
     assert_true_if_warmup_is_equivalent(sampler_old, sampler_new)
 
+# ============ LinearRTO ============
+
+@pytest.mark.parametrize("target", targets)
+def test_LinearRTO_regression_sample(target: cuqi.density.Density):
+    """Test the LinearRTO sampler regression."""
+    sampler_old = cuqi.sampler.LinearRTO(target)
+    sampler_new = cuqi.experimental.mcmc.LinearRTONew(target)
+    assert_true_if_sampling_is_equivalent(sampler_old, sampler_new)
+
+@pytest.mark.parametrize("target", targets)
+def test_LinearRTO_regression_warmup(target: cuqi.density.Density):
+    """Test the LinearRTO sampler regression."""
+    sampler_old = cuqi.sampler.LinearRTO(target)
+    sampler_new = cuqi.experimental.mcmc.LinearRTONew(target)
+    assert_true_if_warmup_is_equivalent(sampler_old, sampler_new)
+
+# ============ RegularizedLinearRTO ============
+def test_RegularizedLinearRTO_regression_sample():
+    """Test the LinearRTO sampler regression."""
+    n = 128
+    A, y_data, info = cuqi.testproblem.Deconvolution1D(dim=n, phantom='square').get_components()
+    x = cuqi.implicitprior.RegularizedGaussian(0.5*np.ones(n), 0.1, constraint = "nonnegativity")
+    y = cuqi.distribution.Gaussian(A@x, 0.001)
+
+    joint = cuqi.distribution.JointDistribution(x, y)
+    target = joint(y=y_data)
+
+    sampler_old = cuqi.sampler.RegularizedLinearRTO(target)
+    sampler_new = cuqi.experimental.mcmc.RegularizedLinearRTONew(target)
+    assert_true_if_sampling_is_equivalent(sampler_old, sampler_new)
```

Comment:
I suggest creating another list of `targets` related to LinearRTO and RegularizedLinearRTO and using pytest parametrize to test against them.

I also suggest adding the `warmup` test even if it seems redundant with  `tune` being pass. It will catch any issues in the future.

Reply:
Just added a function to create a list of "regularized" targets to test with RegularizedLinearRTO. Relevant code is like:
```python
def create_regularized_target(dim=16):
    """Create a regularized target."""
    A, y_data, info = cuqi.testproblem.Deconvolution1D(dim=dim, phantom='square').get_components()
    x = cuqi.implicitprior.RegularizedGaussian(0.5*np.ones(dim), 0.1, constraint = "nonnegativity")
    y = cuqi.distribution.Gaussian(A@x, 0.001)
    return cuqi.distribution.JointDistribution(x, y)(y=y_data)

regularized_targets = [
    create_regularized_target(dim=32),
    create_regularized_target(dim=64),
    create_regularized_target(dim=128)
]
```